docs/tutorials/11_quantum_convolutional_neural_networks.ipynb

```diff
@@ -0,0 +1,1053 @@
CQiCX01gj0Q4cn3lym/sYlkZmbS7oHkVJ6Tl8sjhNjk+GDu1KlT9evXxzz16tUr8wvCkCFDUI930CfR0EZvEptERTJ+/Hi052rVqs2ePZurdPzGJrJq1SrsaJj6yCOPlPOXd7NCiE0YuVHnQOgHMr3169fzNG9HUVRUxBMU0DaEtWvXcpWBvLy8Ro0axcbGUgeCESCWuvXWW2lqUBkZGZgfgRWjPq4KRmKTiBibNm1CYIqKivrkk0+46noBYhMcOXKEktOoUaO4yl4hxCbHB3N9+vSh958yZQpXlUpPT6dJ+rvJhTZ6k9gkKowVK1bgGIz9ceXKlVx1vQCxCX744QdKTpMmTeIqW4QQm3bs2IFFSMuWLblW54033uDJHk+3bt24VkF+fj5Fkxo1auA11/qDsRwSEubUTgDPysryfqBnxowZVBOY9lf069ePq4KR2CQiw/nz5xs2bIjGOnbsWK7yETg2wddff40hIIJXZmYmV9nIbGxyfDCHsS/NgzXHkI5rS2mJ6ptvvuEqrxBGbxKbRMXw66+/0jF1+vTpXOUjcGyCdevWVa1aFcELR3SuCr8QYtOECROwCHn55Ze5tlROTg7lP/Lhhx/yBAVz5syhpe69916uMoC4htnQpWsnbOAFLQs///wzVQZQUlJCPTO6LGRWrg1IYpOIDEOHDkVL7dKlS4DzJYPGJqABULNmzUz9imQJs7HJ8cHcAw88QG/+/PPPc5VO27ZtaSrCKFd5hTB6c3NswlpdvHiRC0IENHDgQLRk7Dhc9idobIJRo0ZhnjvuuMO2MwpCiE2dO3fGImTjxo1c64XsgpXnaR7PkCFDeIKau+++mxYMkD7h8uXLGAZjtldffZWrvDGIVgwjSZXYBD179vR+mgdBkKsCktgkIsDhw4cx9sIe8uOPP3KVPyqxqbi4mAYo77zzDlfZxWxscnYw95///AdjOJrN9/YN2IyIXDS1zBUrIYzeXBibcBjo06cPXbAJeIHgfvz4cZ4shI89e/agzcfFxQVuJyqxqaCggO7xYWq/Lg+zsenChQvok//YN7wWL16cmZm5cOHCSZMm3XPPPVzrvUOK2WsDi4qK6BoUWLduHdcayM7OxmbnQim8w/bt29VvhTVu3Dj6OO1avMAkNokI8Oyzz6KZpqenc9mASmwCOpMavZKpnbn8zMYmZwdzo0ePpnn8nly5e/dumgq+v9+ZHb25KjbhiIWEhOMf/Ql66M3L/CIphIa+aho5ciSXDajEJkBgwmytW7fmcpiZjU1ffvkl5g8A/caoUaMw+uIFlGEX47fweMpcKx0mc+fOpY+LiYkJfCoVkdgk3K64uJi+Vtm3bx9XGVCMTVevXqU7zm3dupWrbGEqNjk7mCssLKQoA34PA++++y5NbdiwIVfpmB29uSo2vfTSS7TyfqEpBv7KU1ROly9fxm4VFRUV4J6xRDE2YT+l/WLv3r1cFU5mY5N2RQhgmDF58uTbb7+dy+X7Lv+TTz7hd/F4zN7KJDSfffYZf57Ho3LHUYlNwu2+/vprtFGVC0oVYxOMGDECc44ZM4bLtjAVm5wdzH3++ec8h8cza9YsdNxl0A3lwO/PfGZHb+6JTVlZWdpPk0aaNGni1D0shGutWbMGbaNDhw5cNqYYmyAtLQ1zvvXWW1wOJ7OxKTk5GfNDlSpVfv31V9To407Pnj1pthAgcvG7+Psmu4ynn366joJFixbxAv7or61ROQ1fYpNwO9qf//KXv3DZmHpsolBy//33c9kWpmKTs4M59J48RzBYMV5Gx+zozT2xiW5GFZTRteWi0po0aRIaxosvvshlY+qxacGCBZjzscce43I4mYpNP/74I2Ym2s+IBQUF2tmWGHsono7tS7vIF28S9OZVTzzxBM0cWOD7jOsvvlm9ejXXGgt7bCouLs7OzuYhqvBn69at/Er489RTT6GNPvvss1w2RpeftGrVisvGKDo0b96cy7bAJ4JibHJ2MNe+fXuaAX3iaz769+9PU8Fv8DI7ervhhhsw5+bNm3lLOeeWW26h1Q4MoZYXEDpbtmzhV5UPwg0axiuvvMJlY3RRcOfOnblsbP78+ZizXbt2XA6np59+Gp+lGJtmzpyJmclLL73EtaUnoZKJEydyrUkUQIn+uQh+5efn/6Yg8M1Z9Gdqrlq1imuN1a5dG3Nu2rSJt135+IlNiqM3ISoJ7MO8bxhzfDCnPXz09ddf5yqd2bNn09S2bdty1fXMjt60B5oKIRyEHob3yYB69erFC3g8a9as4dpr13bt2sW1Hk/Tpk251iR6UgrJzc3l2nDasGEDf57Ho/JITZ7VKvyuOkuWLEFYvkMYaNWqVYMGDZo3b85l4YOyQkJCApeNUSMELhvDBsds1apV47ItaN1UHorn7GDuwoULPPn6PlGTmppKU/FWXHU9s6O3+Ph4zNmyZUveUs6hZxcGpdIaK5vbbrsNXVmLFi24XMnUqVMHDSMpKYnLxqgJAZeN3XzzzZgNYxguhxOtUo8ePXifNIbxm3ZNCQY8ZTo0HNFoEmzZsoVrzUBm4OU9nkOHDnFtOOlPKjhy5AjXGqM5k5OTeduVj5zbJKxH347ob2hmRP3cJroBv8r5mxZSP7fJ2cEcOg6e7PH4PuK3sLCQUg4YdWpmR2/uObdJ+3UysKVLl/ICQnjRVSZvvvkml42pn9u0evVqzHnPPfdwOZzUz23S/wTftWtXri01ffp0nubxDB48mGt1Fi1ahP29jEaNGl26dIlm+ON3q1L6BzcpUhmXlqHdxw4DabkBgagIPvroI7TRBx98kMvG1GMTRQftMUb2UIxNjg/msrOzaSqGuVyloz0/uE2bNlzlw+zoDf0m5nRDbNJ/z2cE/zu+aVJUcnRLjkGDBnHZmHpsmjp1KuZ85plnuBxO6rGJAiLx/c4bI7EY7wPCoWbNmloY0gwYMACTkDwae9F5Qvq7el69epVOdoTFixdzbTBHjx7t27dvYmIilqpbt+6TTz6pfhrD6
```

Comment:
This is just a comment. Here you have chosen pairs 1,3 and 2,4 so as to have output on 3 and 4. Logically you could have chose 1,2 and 3,4 with the output on 2 and 4 right. In the latter the connectivity would be closer for the entangling gates. But I guess at some point as you continue this you would end up with pairs further apart so may still need swaps then, but arguably less since its less pairs. Maybe one can choose some optimal pairing but I guess thats more an advanced consideration for running on a real device.

Reply:
Very up for exploration at the moment, I'm currently looking into different architectures and different unitaries, as well as trying Christas suggestion of dynamic circuits.. A lot to look into so let me know if you have any more suggestions!